### PR TITLE
DBC areas to controller

### PR DESCRIPTION
**Pull request by avengerweb**

### CODE REVIEW

File: app/Http/Controllers/Achievements/AchievementsController.php [hunk 1]
```diff
@@ -31,7 +33,7 @@ public function __construct(TCAPI $api)
      *
      * @param Request $request
      * @param int $id
-     * @return Achievement|array|\Illuminate\Database\Eloquent\Collection|\Illuminate\Database\Eloquent\Model|\Illuminate\Database\Query\Builder|static[]
+     * @return Achievement|array
      */
     public function getAchievements(Request $request, $id = 0) {
```
Comment (xTurinx) [inline app/Http/Controllers/Achievements/AchievementsController.php:38]: You don't need a Request object as argument. 
The global function `request()` returns this object.

File: app/Http/Controllers/Areas/AreasController.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+class AreasController extends Controller
+{
+    private $api;
+
+    /**
+     * AchievementsController constructor.
+     *
+     * @param TCAPI $api
+     */
+    public function __construct(TCAPI $api)
```
Comment (xTurinx) [inline app/Http/Controllers/Areas/AreasController.php:25]: You could make TCAPI to a singleton. So you don't need to initialize it for each controller

File: app/Models/DBC/Area/AreasAndZones.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * @property integer $m_ID
+ * @property integer $m_ParentAreaID
+ * @property string $m_AreaName_lang
+ * @method static \Illuminate\Database\Query\Builder|AreasAndZones whereMID($value)
+ * @method static \Illuminate\Database\Query\Builder|AreasAndZones whereMParentAreaID($value)
+ * @method static \Illuminate\Database\Query\Builder|AreasAndZones whereMAreaNameLang($value)
+ * @mixin \Eloquent
+ */
+class AreasAndZones extends Model
```
Comment (xTurinx) [inline app/Models/DBC/Area/AreasAndZones.php:18]: Maybe we should defined the properties and setter and getter for better IDE autocompletion support

File: app/Http/Controllers/Areas/AreasController.php (hunk 2 above)
Comment (avengerweb) [inline app/Http/Controllers/Areas/AreasController.php:25]: Learn about laravel IOC and see my commit with this class implementation.

File: app/Http/Controllers/Achievements/AchievementsController.php (hunk 1 above)
Comment (avengerweb) [inline app/Http/Controllers/Achievements/AchievementsController.php:38]: Laravel IOC, too and code style.

File: app/Models/DBC/Area/AreasAndZones.php (hunk 3 above)
Comment (avengerweb) [inline app/Models/DBC/Area/AreasAndZones.php:18]: And now, getters and setters isn`t laravel style, u can read about it. If u need getters and setters for params u can read about mutators: https://laravel.com/docs/5.2/eloquent-mutators

File: app/Http/Controllers/Areas/AreasController.php (hunk 2 above)
Comment (xTurinx) [inline app/Http/Controllers/Areas/AreasController.php:25]: I've learned a lot about Laravel and I know that it is possible to do things in different ways which are all correct.

I know what you have done with the TCAPI class but this way of initializing an object makes just sense for e.g. repositories. TCAPI delivers static content so why we don't make a singleton?

Comment (OniBoov) [inline app/Http/Controllers/Areas/AreasController.php]: https://github.com/ShinDarth/TC-JSON-API/commit/cef00b36efffa99afb0b82f6f51964bfee5ad12a#diff-80bda0ca3af455d6354e095203e0199aR25
For what I need write more code? If that already do other people?

Comment (avengerweb) [inline app/Http/Controllers/Areas/AreasController.php:25]: https://github.com/ShinDarth/TC-JSON-API/commit/cef00b36efffa99afb0b82f6f51964bfee5ad12a#diff-80bda0ca3af455d6354e095203e0199aR25

For what I need write more code? If other peoples make it already?

**Pull request merged**